relays/ethereum/src/instances.rs

```diff
@@ -33,83 +33,84 @@ use rialto_runtime::Call;
-					(
-						into_substrate_ethereum_header(&header.header()),
-						into_substrate_ethereum_receipts(header.extra()),
-					)
-				})
-				.collect(),
-		);
-
-		rialto_runtime::Call::BridgeRialto(pallet_call)
-	}
-
-	fn build_unsigned_header_call(&self, header: QueuedEthereumHeader) -> Call {
-		let pallet_call = rialto_runtime::BridgeEthPoACall::import_unsigned_header(
-			into_substrate_ethereum_header(&header.header()),
-			into_substrate_ethereum_receipts(header.extra()),
-		);
-
-		rialto_runtime::Call::BridgeRialto(pallet_call)
-	}
-
-	fn build_currency_exchange_call(&self, proof: Proof) -> Call {
-		let pallet_call = rialto_runtime::BridgeCurrencyExchangeCall::import_peer_transaction(proof);
-		rialto_runtime::Call::BridgeRialtoCurrencyExchange(pallet_call)
-	}
-}
-
-/// Corresponds to the Kovan instance used in the bridge runtime.
-#[derive(Default, Clone, Debug)]
-pub struct Kovan;
-
-impl BridgeInstance for Kovan {
-	fn build_signed_header_call(&self, headers: Vec<QueuedEthereumHeader>) -> Call {
-		let pallet_call = rialto_runtime::BridgeEthPoACall::import_signed_headers(
-			headers
-				.into_iter()
-				.map(|header| {
-					(
-						into_substrate_ethereum_header(header.header()),
-						into_substrate_ethereum_receipts(header.extra()),
-					)
-				})
-				.collect(),
-		);
-
-		rialto_runtime::Call::BridgeKovan(pallet_call)
+impl BridgeInstance {
+	/// Used to build a `Call` for importing signed headers to a Substrate runtime.
+	pub fn build_signed_header_call(&self, headers: Vec<QueuedEthereumHeader>) -> Call {
+		match *self {
+			Self::TestPoA => {
+				let pallet_call = rialto_runtime::BridgeEthPoACall::import_signed_headers(
+					headers
+						.into_iter()
+						.map(|header| {
+							(
+								into_substrate_ethereum_header(&header.header()),
+								into_substrate_ethereum_receipts(header.extra()),
+							)
+						})
+						.collect(),
+				);
+
+				rialto_runtime::Call::BridgeRialto(pallet_call)
+			}
+			Self::Kovan => {
+				let pallet_call = rialto_runtime::BridgeEthPoACall::import_signed_headers(
+					headers
+						.into_iter()
+						.map(|header| {
+							(
+								into_substrate_ethereum_header(header.header()),
+								into_substrate_ethereum_receipts(header.extra()),
+							)
+						})
+						.collect(),
+				);
+
+				rialto_runtime::Call::BridgeKovan(pallet_call)
+			}
+		}
 	}
 
-	fn build_unsigned_header_call(&self, header: QueuedEthereumHeader) -> Call {
-		let pallet_call = rialto_runtime::BridgeEthPoACall::import_unsigned_header(
-			into_substrate_ethereum_header(header.header()),
-			into_substrate_ethereum_receipts(header.extra()),
-		);
-
-		rialto_runtime::Call::BridgeKovan(pallet_call)
+	/// Used to build a `Call` for importing an unsigned header to a Substrate runtime.
+	pub fn build_unsigned_header_call(&self, header: QueuedEthereumHeader) -> Call {
+		match *self {
+			Self::TestPoA => {
+				let pallet_call = rialto_runtime::BridgeEthPoACall::import_unsigned_header(
```

Comment:
Same here, the `pallet_call` seems the duplicated.

Reply:
Reverted